src/wymeditor/editor/native-edit-registration.js

```diff
@@ -19,20 +19,49 @@ WYMeditor.NativeEditRegistration = function (wym) {
     // https:/github.com/mightyiam/edited
     nativeEditRegistration.edited = new WYMeditor.EXTERNAL_MODULES.Edited(
         wym._doc.body,
-        function () {
-            nativeEditRegistration._onNativeEdit.call(nativeEditRegistration);
-        }
+        nativeEditRegistration._onSensibleNativeEdit
+            .bind(nativeEditRegistration),
+        nativeEditRegistration._onAnyNativeEdit
+            .bind(nativeEditRegistration)
     );
 };
 
 /**
-    WYMeditor.NativeEditRegistration._onNativeEdit
-    ==============================================
+    WYMeditor.NativeEditRegistration._onSensibleNativeEdit
+    ======================================================
 
-    Handles native edits for the purpose of Undo/Redo.
+    Handles "sensible" native edits.
+    Sensible according to https://github.com/PolicyStat/edited
 */
-WYMeditor.NativeEditRegistration.prototype._onNativeEdit = function () {
-    var nativeEditRegistration = this;
+WYMeditor.NativeEditRegistration.prototype
+    ._onSensibleNativeEdit = function () {
 
+    var nativeEditRegistration = this;
     nativeEditRegistration.wym.registerModification(true);
 };
+
+/**
+    WYMeditor.NativeEditRegistration._onAnyNativeEdit
+    =================================================
+
+    Handles all native edits.
+*/
+WYMeditor.NativeEditRegistration.prototype._onAnyNativeEdit = function () {
+    var nativeEditRegistration = this,
+        undoRedo = nativeEditRegistration.wym.undoRedo;
+
+    // remove redo points
+    undoRedo.history.changesetsFore = [];
+
+    // Non-native modifications are registered when they are performed.
+    // Contrary to those, only a part of the native edits are
+    // registered when they are performed.
+    // A state with unregistered modifications will be lost upon an undo
+    // and redo because it was never registered.
+    // The last registered state doesn't include it and that will be the result
+    // of the undo and redo operation.
+    // This flag lets the undo operation know that it should add the current
+    // state as a history point (register it) before undoing.
+    // Then the next redo will revert to this state.
```

Comment:
@winhamwr I'm no big fan of the "why" word. I'm a fan of a wide understanding.

To me, most human answers to "why" are narrow points of view.

Trying to answer "why" is usually by asking what could be expected to have been different that would have prevented the outcome that we're asking about.

That's guess work.

I tried to provide just enough background here for a sufficient understanding.

Does it work for you?